systemd/bootupd.service

```diff
@@ -4,7 +4,8 @@ Documentation=https://github.com/coreos/bootupd
 
 [Service]
 Type=notify
-ExecStart=/usr/libexec/bootupd daemon
+Environment=BOOTUPD_VERBOSITY="-v"
+ExecStart=/usr/libexec/bootupd daemon $BOOTUPD_VERBOSITY
```

Comment:
Feels like systemd really needs to grow some sort of
`ExecStartArgs=` that could be added with a drop-in or something; this pattern of bridging environment to args seems somewhat fixed and repetitive.

Wouldn't it be better to support configuration via environment directly?  Can't they already do that via `RUST_LOG`?

Reply:
> Feels like systemd really needs to grow some sort of ExecStartArgs=

We can probably cover this RFE somewhere before going to the systemd folks, but I'm not so sure and I think that the final interface would be quite similar (a map of string -> string) with the only difference being leaving `envp` pristine.

> Wouldn't it be better to support configuration via environment directly? Can't they already do that via RUST_LOG?

Yes `RUST_LOG` would still work, especially for more advanced things (e.g. selectively enabling debug messages from a specific library).
It isn't a very discoverable/friendly UX though, and for `bootupctl` specifically there are few bleeding edges as the crate name (used to control per-module logging) does not match the binary name.
This is also narrower in scope, as it only bridges the top-level crate level (versus the global logging level) to a CLI flag, allowing to directly get debug/trace messages from the application without all the transitive dependencies too (i.e. `RUST_LOG=bootupd=debug`).
To me it seems worth at least for `bootupctl`, and possibly on `bootupd` to avoid having to each every single user about the module-syntax of `RUST_LOG`.

Reply:
I feels to me that this would be rarely used and maybe best left to a doc page with general debugging instructions.

Reply:
But this is similar to what you did in [incati](https://github.com/coreos/zincati/blob/master/dist/systemd/system/zincati.service) and there is value in being consistent so I don't know.